examples/perfmanager/main.go

```diff
@@ -0,0 +1,111 @@
+/*
+Copyright (c) 2017 VMware, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+This example program shows how the `view` and `property` packages can
+be used to navigate a vSphere inventory structure using govmomi.
+*/
+
+package main
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/vmware/govmomi/examples"
+	"github.com/vmware/govmomi/find"
+	"github.com/vmware/govmomi/performance"
+	"github.com/vmware/govmomi/vim25"
+	"github.com/vmware/govmomi/vim25/types"
+)
+
+func main() {
+	examples.Run(func(ctx context.Context, c *vim25.Client) error {
+		// Find virtual machine list
+		f := find.NewFinder(c, true)
+
+		dc, err := f.DefaultDatacenter(ctx)
+		if err != nil {
+			return err
+		}
+
+		f.SetDatacenter(dc)
+
+		vms, err := f.VirtualMachineList(ctx, "*")
+		if err != nil {
+			return err
+		}
+
+		// Convert virtual machines into list of references
+		var vmsRefs []types.ManagedObjectReference
+		for _, vm := range vms {
+			vmsRefs = append(vmsRefs, vm.Reference())
+		}
+
+		// Create a PerfManager
+		perfManager := performance.NewManager(c)
+
+		// Retrieve counters name list
+		counters, err := perfManager.CounterInfoByName(ctx)
+		if err != nil {
+			return err
+		}
+
+		var names []string
+		for name := range counters {
+			names = append(names, name)
+		}
+
+		// Create PerfQuerySpec
+		spec := types.PerfQuerySpec{
+			MaxSample:  1,
+			MetricId:   []types.PerfMetricId{{Instance: "*"}},
+			IntervalId: 300,
+		}
+
+		// Query metrics
+		sample, err := perfManager.SampleByName(ctx, spec, names, vmsRefs)
+		if err != nil {
+			return err
+		}
+
+		result, err := perfManager.ToMetricSeries(ctx, sample)
+		if err != nil {
+			return err
+		}
+
+		// Read result
+		for _, metric := range result {
+			name := metric.Entity
+
+			for _, v := range metric.Value {
+				counter := counters[v.Name]
+				units := counter.UnitInfo.GetElementDescription().Label
+
+				instance := v.Instance
+				if instance == "" {
+					instance = "-"
+				}
+
+				if v.ValueCSV() != "" {
```

Comment:
Better to just call `ValueCSV` once as it does some work.  You could assign to a var first or change the check to:
```go
if len(v.Value) != 0 {
```

Reply:
Changed. :)